### PR TITLE
Add ability to customize normals provided to render_offscren_rgb

### PR DESCRIPTION
**Pull request by jabooth**
This is useful for allowing for per-triangle per-vertex interpolations (e.g. for inverse rendering applications).

### CODE REVIEW

File: cyrasterize/glrasterizer.pyx [hunk 1]
```diff
@@ -104,6 +104,32 @@ def normalize_v3(arr):
...
     return arr
 
+
+def vertex_normals(points, trilist):
+    # Create a zeroed array with the same type and shape as our vertices i.e., per vertex normal
+    cdef np.ndarray[float, ndim=2, mode="c"] norm = np.zeros( (points.shape[0], points.shape[1]), dtype=np.float32)
+    tris = points[trilist]
+
+    # Calculate the normal for all the triangles, by taking the cross product of the vectors v1-v0, and v2-v0 in each triangle
+    n = np.cross( tris[::,1 ] - tris[::,0]  , tris[::,2 ] - tris[::,0] )
```
Comment (patricksnape) [inline cyrasterize/glrasterizer.pyx]: I have a fast Cython based cross product in menpo that I was finding to be about 20% faster than numpy.

**Comment by patricksnape** [comment]
Fine, don't listen to me. 😢

File: cyrasterize/glrasterizer.pyx (hunk 1 above)
Comment (jabooth) [inline cyrasterize/glrasterizer.pyx]: sorry @patricksnape missed this!

Yeah I know, could be brought over for sure.

To be clear this only moves the existing normals calc into a function, did the timings and our vertex normals  calc is faster for sure. For now just focused on getting this together for release so we can make progress with the 3DMM project.

**Comment by jabooth** [comment]
@patricksnape See comments above and #11!

**Pull request merged**